python/py-rawpy/Portfile

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           github 1.0
+
+name                py-rawpy
+version             0.17.2
+revision            0
+license             MIT
+
+github.setup        letmaik rawpy v0.17.2
+fetch.type          git
+
+python.versions     310
+
+maintainers         openmaintainer @jsalort
+categories-append   graphics
+
+description         Python wrapper for the LibRaw library
+long_description    rawpy is an easy-to-use Python wrapper for the LibRaw library. It also contains some extra \
+                    functionality for finding and repairing hot/dead pixels.
+
+homepage            https://github.com/letmaik/rawpy
+
+patchfiles          setup.py.diff
+
+if {${name} ne ${subport}} {
+
+    depends_lib-append \
+                    port:libraw \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-cython \
+                    port:py${python.version}-scikit-image
+
+    depends_test-append \
+                    port:py${python.version}-pytest \
+                    port:py${python.version}-imageio
+
+    configure.cflags-append "-I${prefix}/include/libraw"
+    configure.cxxflags-append "-I${prefix}/include/libraw"
```

Comment:
this doesn't appear to work as the CI fails to find the headers

Reply:
I don't understand why it is different on the CI and locally on my machine.

On the CI, there is
```bash
/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk -arch x86_64 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk -D_HAS_LIBRAW_CONFIG_H=0 -Irawpy -I/opt/local/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/numpy/core/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10 -c rawpy/_rawpy.cpp -o build/temp.macosx-11.0-x86_64-cpython-310/rawpy/_rawpy.o
```

On my machine, there is
```bash
/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -arch arm64 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -D_HAS_LIBRAW_CONFIG_H=0 -Irawpy -I/opt/local/include/libraw -I/opt/local/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/numpy/core/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10 -c rawpy/_rawpy.cpp -o build/temp.macosx-12.0-arm64-cpython-310/rawpy/_rawpy.o
```

So `-I/opt/local/include/libraw` is correctly added on my machine, but not on the CI.
I tried using cppflags-append instead of cflags-append and cxx-flags-append, as was suggested by Joshua Root for the libraw port. It makes no difference (it works on my machine and not on the CI). I don't suppose that my machine being arm64 makes any difference in this matter?

Reply:
You don't have a configure phase with the python portgroup, so `configure.*` options aren't used. It might be easiest to patch the needed flags into the `setup.py`.

Reply:
Ah, I wasn't aware that configure.* options were useless here. With that in mind, I figured that the reason it was working locally is because pkgconfig is installed. Indeed, the setup.py script uses pkgconfig to add CFLAGS. I tried in a virtual machine with blank MacPorts installation and I checked that adding pkgconfig as a build dependency fixes the issue.

However, the CI still fails, but now very early on, after 28 seconds, because /tmp/mpbb cannot be found. Is there some unrelated problems with the CI ?

Reply:
Yes, there was an issue with CI which has since been resolved.